cborg/tests/Tests/Properties.hs

```diff
@@ -399,20 +399,26 @@ prop_decodeRefdecodeImp _ x =
 -- >                Imp ─────────────────────▶Imp
 -- >                              id
 --
--- > (fromFlatTerm dec_imp . toFlatTerm . enc_imp) imp = Right imp
+-- > (fromFlatTerm dec_imp . toFlatTerm . enc_imp) imp = imp
 --
 prop_toFromFlatTerm :: forall t. Token t => Proxy t -> t -> Bool
 prop_toFromFlatTerm _ x =    
   
-    liftA2 eq (fn enc) (Right imp) == Right True
+  (deserialiseFromFlatTerm (decodeImp t) . toFlatTerm . encodeImp t) imp  `eq`  imp
```

Comment:
You should probably use QuickCheck's `===` operator there because if the property fails `===` gives a much better error message.

Reply:
Here, `eq` is not directly equivalent to QC's `===` as `eq` is defined as `eq = eqImp t` and `eqImp` is a method of TypeClass `Token` for which the test property is written.

Reply:
Ah, ok. Disregard that comment then.